inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/eltwise/eltwise_kernel_b_fs_yx_fsv16.cpp

```diff
@@ -61,18 +63,32 @@ JitConstants EltwiseKernel_b_fs_yx_fsv16::MakeLoadJitConstants(const eltwise_par
                 case EltwiseInputMode::INPUT_BUFFER:
                     if (params.inputs[input.index].LogicalSize() == params.output.Feature().v &&
                         params.inputs[input.index].LogicalSize() == params.inputs[input.index].Feature().v) {
-                        jit.AddConstant(MakeJitConstant(name,
+                        if (DataTensor::ChannelsCount(params.output.GetLayout()) == 4) {
+                            jit.AddConstant(MakeJitConstant(name,
                                                         "BLOCK_READN(INPUT" + std::to_string(input.index) + "_TYPE, 1, input" + std::to_string(input.index) +
                                                         ", INPUT"+std::to_string(input.index)+"_GET_INDEX(b, f_block*16, y, x))"));
+                        } else {
+                            jit.AddConstant(MakeJitConstant(name,
+                                                        "BLOCK_READN(INPUT" + std::to_string(input.index) + "_TYPE, 1, input" + std::to_string(input.index) +
+                                                        ", INPUT"+std::to_string(input.index)+"_GET_INDEX(b, f_block*16, z, y, x))"));
+                        }
                     } else if (params.inputs[input.index].LogicalSize() == 1) {
                         jit.AddConstant(MakeJitConstant(name,
                                                         "input" + std::to_string(input.index) +
                                                         "[0]"));
                     } else {
-                        std::string block_read_str = "BLOCK_READN(INPUT" + std::to_string(input.index) + "_TYPE, " +
-                                                                 "BLOCK_SIZE, " +
-                                                                 "input" + std::to_string(input.index) + ", " +
-                                                                 "INPUT" + std::to_string(input.index) + "_GET_INDEX(b, f_block*16, y, x))";
+                        std::string block_read_str = "";
+                        if (DataTensor::ChannelsCount(params.output.GetLayout()) == 4) {
+                            block_read_str = "BLOCK_READN(INPUT" + std::to_string(input.index) + "_TYPE, " +
+                                             "BLOCK_SIZE, " +
+                                             "input" + std::to_string(input.index) + ", " +
+                                             "INPUT" + std::to_string(input.index) + "_GET_INDEX(b, f_block*16, y, x))";
+                        } else {
+                            block_read_str = "BLOCK_READN(INPUT" + std::to_string(input.index) + "_TYPE, " +
+                                             "BLOCK_SIZE, " +
+                                             "input" + std::to_string(input.index) + ", " +
+                                             "INPUT" + std::to_string(input.index) + "_GET_INDEX(b, f_block*16, z, y, x))";
```

Comment:
You can simplify this via replacement of common code before if because only _GET_INDEX part is different.